controllers/argocd/route.go

```diff
@@ -308,54 +308,75 @@ func (r *ReconcileArgoCD) reconcileApplicationSetControllerWebhookRoute(cr *argo
 
 	// Allow override of the Annotations for the Route.
 	if len(cr.Spec.ApplicationSet.WebhookServer.Route.Annotations) > 0 {
-		route.Annotations = cr.Spec.Server.Route.Annotations
+		route.Annotations = cr.Spec.ApplicationSet.WebhookServer.Route.Annotations
 	}
 
 	// Allow override of the Labels for the Route.
 	if len(cr.Spec.ApplicationSet.WebhookServer.Route.Labels) > 0 {
 		labels := route.Labels
-		for key, val := range cr.Spec.Server.Route.Labels {
+		for key, val := range cr.Spec.ApplicationSet.WebhookServer.Route.Labels {
 			labels[key] = val
 		}
 		route.Labels = labels
 	}
 
 	// Allow override of the Host for the Route.
-	if len(cr.Spec.Server.Host) > 0 {
+	if len(cr.Spec.ApplicationSet.WebhookServer.Host) > 0 {
 		route.Spec.Host = cr.Spec.ApplicationSet.WebhookServer.Host
 	}
 
-	if cr.Spec.Server.Insecure {
-		// Disable TLS and rely on the cluster certificate.
-		route.Spec.Port = &routev1.RoutePort{
-			TargetPort: intstr.FromString("webhook"),
+	route.Spec.Port = &routev1.RoutePort{
+		TargetPort: intstr.FromString("webhook"),
+	}
+
+	// Allow override of TLS options for the Route
+	if cr.Spec.ApplicationSet.WebhookServer.Route.TLS != nil {
+		tls := &routev1.TLSConfig{}
+
+		// Set Termination
+		if cr.Spec.ApplicationSet.WebhookServer.Route.TLS.Termination != "" {
+			tls.Termination = cr.Spec.ApplicationSet.WebhookServer.Route.TLS.Termination
+		} else {
+			tls.Termination = routev1.TLSTerminationPassthrough
```

Comment:
I think this should default to TLSTerminationEdge, because the ApplicationSet webhook does not support [HTTPS AFAICT](https://github.com/argoproj/argo-cd/blob/3c2124235619d8451e2d24c7873e5a6da17354af/cmd/argocd-applicationset-controller/commands/applicationset_controller.go#L271).

If we instead use `TLSTerminationPassthrough`, then OpenShift will pass the HTTPS connection data directly to the applicationset controller webhook, which will throw it away because it's not valid HTTP.

Reply:
This is done.

Reply:
:+1: